src/registry.rs

```diff
@@ -0,0 +1,47 @@
+use zbus::zvariant;
+
+use crate::{proxy::Proxy, AppID, Error};
+
+#[derive(Debug, zvariant::SerializeDict, zvariant::Type)]
+#[zvariant(signature = "dict")]
+struct RegisterOptions {}
+
+struct RegistryProxy<'a>(Proxy<'a>);
+
+impl<'a> RegistryProxy<'a> {
+    pub async fn new() -> Result<RegistryProxy<'a>, Error> {
+        let proxy = Proxy::new_desktop("org.freedesktop.host.portal.Registry").await?;
+        Ok(Self(proxy))
+    }
+
+    pub async fn register(&self, app_id: AppID) -> Result<(), Error> {
+        let options = RegisterOptions {};
+        self.0.call_method("Register", &(&app_id, &options)).await?;
+        return Ok(());
+    }
+}
+
+impl<'a> std::ops::Deref for RegistryProxy<'a> {
+    type Target = zbus::Proxy<'a>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+/// Registers a host application for portal usage.
+///
+/// Portals rely on the application ID to store and manage the permissions of
+/// individual apps. However, for non-sandboxed (host) applications, this
+/// information cannot be directly retrieved. To resolve this, the method first
+/// verifies that the application is not sandboxed. It then uses the
+/// `org.freedesktop.host.portal.Registry` interface to register the process
+/// communicating over DBus with the portal as the owner of the specified
+/// application ID. For more technical details, see ...
```

Comment:
On thing I am missing is when to call this, e.g. we had problems with the camera permission if it was called too soon. cc @GeorgesStavracas

Reply:
Also the `...` should be replaced before merging.